ethexe/runtime/common/src/journal.rs

```diff
@@ -54,9 +54,8 @@ impl<S: Storage> NativeJournalHandler<'_, S> {
                         stash.add_to_program(dispatch, expiry);
                     });
                 } else {
-                    state
-                        .canonical_queue
-                        .modify_queue(storage, |queue| queue.queue(dispatch));
+                    let queue = state.queue_from_origin(dispatch.origin);
```

Comment:
or something like this 
```suggestion
                    let queue = state.queue_from_origin(dispatch.message_type);
```

Reply:
Fixed